### PR TITLE
Terminal error message

### PR DESCRIPTION
**Pull request by shivani170**
# Description

1. For Unauthorised users, we are showing "connecting.." always on UI

Screenshot 2022-09-09 at 4 28 18 PM

Fixes  https://github.com/devtron-labs/devtron/issues/2280

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
1. Remove all the access
4. Go to terminal

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by vivek-devtron — APPROVED** [review]

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx [hunk 1]
```diff
@@ -288,9 +292,18 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
...
+            .catch((err) => {
+              console.log(err)
+                showError(err)
+                if (err instanceof ServerErrors && Array.isArray(err.errors)) {
+                    const _invalidNameErr = err.errors[0].userMessage
+                    if (_invalidNameErr.includes('Unauthorized')) {
+                        setErrorMessage(Error_Message.UNAUTHORIZED)
+                    }
+                } else{
+                  setErrorMessage(err)
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:304]: Error is an object but ` setErrorMessage` holds string

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx [hunk 2]
```diff
@@ -288,9 +292,18 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
                     postInitialize(sessionId)
                 }
             })
-            .catch((error) => {
-                console.log('error while getNewSession ', error);
-            });
+            .catch((err) => {
+              console.log(err)
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:296]: Remove this as we're showing the same in toast with `showError`

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/node.type.ts [hunk 3]
```diff
@@ -3,4 +3,8 @@ export enum SocketConnectionType {
   CONNECTED = 'CONNECTED',
   DISCONNECTING = 'DISCONNECTING',
   DISCONNECTED = 'DISCONNECTED',
+}
+
+export enum Error_Message {
+  UNAUTHORIZED = 'Not authorized. You do not have permission to access the terminal for this application.'
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/node.type.ts:9]: Change this to `const`

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx [hunk 4]
```diff
@@ -33,6 +34,7 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
     const [firstMessageReceived, setFirstMessageReceived] = useState(false);
     const [popupText, setPopupText] = useState<boolean>(false);
     const isOnline = useOnline();
+    const [errorMessage, setErrorMessage] = useState<string>(undefined)
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:37]: Don't init states with undefined use blank for string type & null for an object.

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx [hunk 5]
```diff
@@ -140,7 +142,9 @@ function TerminalView(terminalViewProps: TerminalViewProps) {
 
         _socket.onerror = function (evt) {
             disableInput();
+            setErrorMessage('Selected command type is not supported. Please select another.')
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:145]: move this to same const mentioned above

File: src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/terminal/Terminal.tsx:304]: Also, confirm about the message

**Review by sohel-rp — APPROVED** [review]

**Pull request merged**